ravenframework/SupervisedLearning/FeatureSelection/VarianceThreshold.py

```diff
@@ -0,0 +1,102 @@
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    spec = super().getInputSpecification()
+    spec.description = r"""The \xmlString{VarianceThreshold} is a feature selector that removes all low-variance features.
+                           This feature selection algorithm looks only at the features and not the desired outputs. The
+                           variance threshold can be set by the user."""
+    spec.addSub(InputData.parameterInputFactory('threshold',contentType=InputTypes.FloatType,
+        descr=r"""Features with a training-set variance lower than this threshold
+                  will be removed. The default is to keep all features with non-zero
+                  variance, i.e. remove the features that have the same value in all
+                  samples.""", default=0.0))
+    return spec
+
+  def __init__(self):
+    super().__init__()
+    self.threshold = 0.0
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+    nodes, notFound = paramInput.findNodesAndExtractValues(['threshold'])
+    assert(not notFound)
+    self.threshold = nodes['threshold']
+    if self.threshold <= 0:
+      raise self.raiseAnError(ValueError, '"threshold" parameter must be > 0' )
+
+  def _train(self, X, y, featuresIds, targetsIds, maskF = None, maskT = None):
+    """
+      Train the feature selection model and perform search of best features
+      @ In, X, numpy.array, feature data (nsamples,nfeatures) or (nsamples, nTimeSteps, nfeatures)
+      @ In, y, numpy.array, target data (nsamples,nTargets) or (nsamples, nTimeSteps, nTargets)
+      @ In, featuresIds, list, list of features
+      @ In, targetsIds, list, list of targets
+      @ In, maskF, optional, np.array, indeces of features to search within
+                                       (parameters to include None if search is whitin targets)
+      @ In, maskT, optional, np.array, indeces of targets to search within
+                                       (parameters to include None if search is whitin features)
+
+      @ Out, newFeatures or newTargets, list, list of new features/targets
+      @ Out, supportOfSupport_, np.array, boolean mask of the selected features
+      @ Out, whichSpace, str, which space?
+      @ Out, vals, dict, dictionary of new values
+    """
+    # if time dependent, we work on the expected value of the features
+    if self.whichSpace == 'feature':
+      space = X if len(X.shape) < 3 else X.mean(axis=(1))
+    else:
+      space = y if len(y.shape) < 3 else y.mean(axis=(1))
+    estimator = vt.fit(space)
+    estimator.get_support()
```

Comment:
There is no return here. There are several outputs listed in the docstring, I guess a return is required here.

Reply:
fixed